src/uniform.js

```diff
@@ -228,6 +228,64 @@ vgl.modelViewUniform = function (name) {
 
 inherit(vgl.modelViewUniform, vgl.uniform);
 
+///////////////////////////////////////////////////////////////////////////////
+/**
+ * Create new instance of class modelViewOriginUniform.
+ *
+ * @param name
+ * @param uniform: a triplet of floats.
+ * @returns {vgl.modelViewUniform}
+ */
+///////////////////////////////////////////////////////////////////////////////
+vgl.modelViewOriginUniform = function (name, origin) {
+  'use strict';
+
+  if (!(this instanceof vgl.modelViewOriginUniform)) {
+    return new vgl.modelViewOriginUniform(name, origin);
+  }
+
+  if (name.length === 0) {
+    name = 'modelViewMatrix';
+  }
+
+  var m_origin = vec3.fromValues(origin[0], origin[1], origin[2]);
+
+  vgl.uniform.call(this, vgl.GL.FLOAT_MAT4, name);
+
+  this.set(mat4.create());
+
+  /////////////////////////////////////////////////////////////////////////////
+  /**
+   * Update the uniform given a render state and shader program.  This offsets
+   * the modelViewMatrix by the origin, and, if the model view should be
+   * aligned, aligns it appropriately.  The alignment must be done after the
+   * origin offset to maintain precision.
+   *
+   * @param {vgl.renderState} renderState
+   * @param {vgl.shaderProgram} program
+   */
+  /////////////////////////////////////////////////////////////////////////////
+  this.update = function (renderState, program) {
+    program = program; /* unused parameter */
```

Comment:
Perhaps we should refactor the uniforms on a different branch then, since all of the uniforms receive program in their update method and none of them use it.